sbt_common/internal_model/src/main/scala/uk/ac/wellcome/models/work/internal/IdentifierType.scala

```diff
@@ -0,0 +1,83 @@
+package uk.ac.wellcome.models.work.internal
+
+import java.io.InputStream
+
+import uk.ac.wellcome.exceptions.GracefulFailureException
+
+import scala.io.Source
+
+case class IdentifierType(
+  id: String,
+  label: String,
+  ontologyType: String = "IdentifierType"
+)
+
+case object IdentifierType {
+  private val stream: InputStream = getClass
+    .getResourceAsStream("/identifier-schemes.csv")
+
+  // identifier-schemes.csv is a CSV file with three columns per row:
+  //
+  //    CALMRefNo,calm-ref-no,CALM ref no
+  //
+  // The first entry is an immutable platform identifier.  The second
+  // and third entries are the ID and label we show in public ID schemes.
+  //
+  // Internally, we use the platform identifier ("CALMRefNo") -- this won't
+  // change even if the ID ("calm-ref-no") or label ("CALM ref no") do.
+  //
+  private val csvRows: Array[(String, String, String)] = Source
+    .fromInputStream(stream)
+    .mkString
+    .split("\n")
+    .map { row =>
+      val columns = row.split(",").map(_.trim)
+      assert(columns.length == 3)
+      (columns(0), columns(1), columns(2))
+    }
+
+  private val identifierTypeMap = csvRows
+    .map {
+      case (platformId, schemeId, schemeLabel) =>
+        Map(
+          platformId -> IdentifierType(
+            id = schemeId,
+            label = schemeLabel
+          )
+        )
+    }
+    .fold(Map[String, IdentifierType]()) { (x, y) =>
+      x ++ y
+    }
+
+  private def lookupPlatformId(id: String): IdentifierType =
+    identifierTypeMap.get(id) match {
+      case Some(identifierType) => identifierType
+      case None =>
+        throw GracefulFailureException(
+          new RuntimeException(s"Unrecognised identifier type: [$id]")
+        )
+    }
+
+  // This isn't ideal, because we're duplicating the list from the CSV,
+  // but it gives us a bit of type-safety.
+  //
+  // If any of these values don't resolve, we fail immediately upon
+  // startup -- not when we try to load the value!
+  //
```

Comment:
I agree with the comment here, I'm not sure what duplicating the entries from the CSV gives us.  If there is a mistake or omission it will fail at runtime, which is the same as if we load the CSV at instantiation and trust it.  For the larger files (e.g., location-types) this seems to be the best option anyway.

Reply:
If we are going down the route of reading the csv files, being fully dynamic and trusting that they are correct might be ok. 
This check at startup basically forces us to duplicate the information in the csv file and if we do that we could just as well duplicate it by keeping the enum and updating the list of allowed values in the enum.
I think especially for big csv like the location-types one, this check is not really a viable option (nor is the enum) and in that case, we are forced to trust those files anyway. 
Also, it's very easy to forget to add new things to these lines as they get added to the csv file, so there is actually no real certainty that the application doesn't start if the csv file is somehow invalid.

Reply:
It fails at runtime, but it fails as soon as you instantiate this object, which I think happens at import, i.e. startup time?

As opposed to failing when you do the query.

If you forget to add an identifier type from the CSV to this list, you can’t used it elsewhere in the codebase, so I don’t think that’s an issue.

The alternative is:

* Everywhere we use an instance of IdentifierSchemes right now, we have to do `lookupPlatformId("MiroImageNumber")` with a hard-coded string
* Something with Shapeless that generates these values for us (but then IntelliJ might not be very happy). Open to suggestions.

Reply:
I don't think you can do it with shapeless. You probably can with macros but I don't think we want to go down that route. 

I don't think this gives us many advantages over just keeping the enum up to date and it's not a viable option for csvs that change often and/or are too big.
I know that this would fail at startup rather than while processing a message. I just don't think that failure while processing the message would be that bad if it give us the ability to just load the content of this file into a map, wihtout having duplicated information.

Reply:
So expose `lookupPlatformId` as a public (and renamed) method, use that everywhere, and drop the constants below?

Reply:
I think so, but I'd like @kenoir and @mklander opinion

Reply:
I think it’s a Singleton, yes.